OpenFeature/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt

```diff
@@ -1,19 +1,34 @@
 package dev.openfeature.sdk
 
 import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 
 object OpenFeatureAPI {
     private var provider: FeatureProvider? = null
     private var context: EvaluationContext? = null
     var hooks: List<Hook<*>> = listOf()
         private set
 
-    suspend fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) = coroutineScope {
+    suspend fun setProviderAsync(
+        provider: FeatureProvider,
+        initialContext: EvaluationContext? = null
+    ) = coroutineScope {
         provider.initialize(initialContext ?: context)
         this@OpenFeatureAPI.provider = provider
         if (initialContext != null) context = initialContext
     }
 
+    fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
+        runBlocking {
+            launch {
```

Comment:
I was thinking this would be a non-blocking `launch`, so that `provider.initialize()` is executed in the background and synchronous `setProvider` returns right away.

Nevertheless, I also see that having both `setProviderAsync` and `setProvider` is not very useful, since a "suspend" function can always be executed as blocking or as a non-blocking background execution. In other words, even if we had the Events functionality in place (prototype [here](https://github.com/spotify/openfeature-kotlin-sdk/pull/32)), we could still keep `setProvider` a suspend function for extra flexibility of the APIs.

Reply:
It can be a non-blocking initialisation of a coroutine, but it would essentially be the same as a suspend function - which I honestly don't see a point of duplicating.
Let's wait for more feedback from OF community!

Reply:
> I was thinking this would be a non-blocking `launch`, so that `provider.initialize()` is executed in the background and the synchronous `setProvider` returns right away.
> 
> Nevertheless, I also see that having both `setProviderAsync` and `setProvider` is not very useful, since a _suspend_ function can always be executed as blocking or as a non-blocking background execution. In other words, even if we had the Events functionality in place (prototype [here](https://github.com/spotify/openfeature-kotlin-sdk/pull/32)), we could still keep `setProvider` a _suspend_ function for extra flexibility of the APIs.

This makes sense to me.

Essentially, the `setProvider` is intended to be a mutator with side effects. It's important to note, that the persona setting the provider may not be the persona evaluating feature-flags (see here for our list of [personas](https://openfeature.dev/specification/glossary#user-roles)). It's quite likely that the configuration code which sets the provider doesn't need to perform any feature-flag evaluations, and therefor it's not "interested" in waiting for a blocking `setProvider` call.

It's the code actually doing feature flag evaluations that must be certain the system is in a ready state. This is why a non-blocking `setProvider` + ready events is the design we chose.

All that said, I understand this might not ideal to Kotlin users. I am open to any solution that allows us to set a provider, and provide a mechanism for all clients (pre-existing or not) to "know" the system is in a ready state.